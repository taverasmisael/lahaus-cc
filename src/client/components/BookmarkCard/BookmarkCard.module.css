.container {
  @apply relative;
}

.empty {
  @apply inline-block w-full h-auto;
}

.title {
  @apply font-bold text-lg mb-2 mt-4;
}
.description {
  @apply text-sm font-light;
}
