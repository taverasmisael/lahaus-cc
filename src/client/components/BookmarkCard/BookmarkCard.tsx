import Image from 'next/image'
import Link from 'next/link'

import styles from './BookmarkCard.module.css'
import emptyStateImg from '../../assets/empty-state.png'
import { RealEstate } from '@/shared/real-state.interface'

function BookmarkCard({ items, name, id }: BookmarkCardProps) {
  console.log(items)
  return (
    <div className={styles.container}>
      <header className={styles.header}>
        {!items.length ? (
          <Image
            src={emptyStateImg}
            alt=""
            layout="responsive"
            className={styles.empty}
          />
        ) : (
          <div>Cute carousel</div>
        )}
      </header>
      <h2 className={styles.title}>
        <Link href={`bookmarks/${id}`}>
          <a>{name}</a>
        </Link>
      </h2>
      <p className={styles.description}>{getDescriptionText(items)}</p>
    </div>
  )
}

interface BookmarkCardProps {
  items: RealEstate[]
  name: string
  id: string
}

function getDescriptionText(items: any[]) {
  return items.length === 1
    ? '1 propiedad guardada'
    : `${items.length} propiedades guardadas`
}

export default BookmarkCard
