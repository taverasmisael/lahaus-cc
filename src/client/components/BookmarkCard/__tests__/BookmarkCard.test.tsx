import { RealEstate, RealEstateAttributes } from '@/shared/real-state.interface'
import { render } from '@testing-library/react'

import BookmarkCard from '../BookmarkCard'

const DEFAULT_ITEMS = [
  {
    id: '1',
    attributes: {
      gallery_urls: ['https://unsplash.it/1'],
    },
  },
  {
    id: '2',
    attributes: {
      gallery_urls: ['https://unsplash.it/2'],
    },
  },
] as RealEstate[]

describe('components/BookmarkCard', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BookmarkCard id="3" items={[]} name="My empty bookmark" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('renders correctly one item', () => {
    const { container } = render(
      <BookmarkCard id="3" items={[DEFAULT_ITEMS[0]]} name="My bookmarks" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('renders correctly multi items', () => {
    const { container } = render(
      <BookmarkCard id="3" items={DEFAULT_ITEMS} name="My bookmarks" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
