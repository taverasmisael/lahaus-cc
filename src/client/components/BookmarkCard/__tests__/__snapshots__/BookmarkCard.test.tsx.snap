// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BookmarkCard renders correctly 1`] = `
<div
  class="container"
>
  <header
    class="header"
  >
    <div
      class="carousel singleChild"
    >
      <div
        class="image image0"
      >
        <img
          class="w-full h-full object-cover"
          src="/imgs/empty-state.png"
        />
      </div>
    </div>
  </header>
  <h2
    class="title"
  >
    <a
      href="/bookmarks/3"
    >
      My empty bookmark
    </a>
  </h2>
  <p
    class="description"
  >
    0 propiedades guardadas
  </p>
</div>
`;

exports[`components/BookmarkCard renders correctly multi items 1`] = `
<div
  class="container"
>
  <header
    class="header"
  >
    <div
      class="carousel twoChildren"
    >
      <div
        class="image image0"
      >
        <img
          class="w-full h-full object-cover"
          src="https://unsplash.it/1"
        />
      </div>
      <div
        class="image image1"
      >
        <img
          class="w-full h-full object-cover"
          src="https://unsplash.it/2"
        />
      </div>
    </div>
  </header>
  <h2
    class="title"
  >
    <a
      href="/bookmarks/3"
    >
      My bookmarks
    </a>
  </h2>
  <p
    class="description"
  >
    2 propiedades guardadas
  </p>
</div>
`;

exports[`components/BookmarkCard renders correctly one item 1`] = `
<div
  class="container"
>
  <header
    class="header"
  >
    <div
      class="carousel singleChild"
    >
      <div
        class="image image0"
      >
        <img
          class="w-full h-full object-cover"
          src="https://unsplash.it/1"
        />
      </div>
    </div>
  </header>
  <h2
    class="title"
  >
    <a
      href="/bookmarks/3"
    >
      My bookmarks
    </a>
  </h2>
  <p
    class="description"
  >
    1 propiedad guardada
  </p>
</div>
`;
